pkg/api/validation/events.go

```diff
@@ -25,11 +25,19 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
We are not enforcing it, but practically speaking all references need to be qualified sufficiently to unambiguously locate something.  Version and kind tuple is a decent way to do that.


> On Feb 17, 2015, at 4:51 PM, Deyuan Deng <notifications@github.com> wrote:
> 
> In pkg/api/validation/events.go:
> 
> >  	if !util.IsDNSSubdomain(event.Namespace) {
> >  		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
> >  	}
> > +	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
> Ah, right. Added a comment.
> 
> Are we enforing version match between api resource and its reference? i.e. is this valid
> {
> kind: "event"
> apiversion: "v1beta1"
> ...
> involvedObj: {
> kind: Node
> apiversion: v1beta3
> }
> }
> 
> If this is invalid, then things might be eaiser since we can do a inline conversion in the event; otherwise we need to do nested conversion?
> 
> —
> Reply to this email directly or view it on GitHub.
>